File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {

        stack<char>st;
        for(char ch : s)
        {
            if(ch == '(' || ch == '[' || ch == '{')
                st.push(ch);
            else if(!st.empty()){
                if(st.top() == '(' && ch!=')')
                    return false;
                if(st.top() == '[' && ch!=']')
                    return false;
                if(st.top() == '{' && ch!='}')
                    return false;
                st.pop();
            }
            else
                return false;
        }
        return st.empty();
    }
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        charSet = []
        mappings = {')' : '(' , ']':'[' , '}' : '{'}
        for ch in s:
            if ch in mappings:
                topElement = charSet.pop() if charSet else '#'
                if mappings[ch]!= topElement:
                    return False
            else:
                charSet.append(ch)
        return len(charSet) == 0    
        

                